### PR TITLE
fix this mornings errors

### PR DESCRIPTION
**Pull request by xjdr**

### CODE REVIEW

File: src/main/java/com/xjeffrose/chicago/db/EncryptedStorageProvider.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  private void configure() {
+    properties.setProperty("chimera.crypto.cipher.classes", "com.intel.chimera.crypto.OpensslCipher");
+    try {
+      cipher = Utils.getCipherInstance(CipherTransformation.AES_CTR_NOPADDING, properties);
+    } catch (IOException e) {
+      log.error("Error while configuring the encryption handler: ", e);
+    }
+  }
+
+  public byte[] encrypt(byte[] raw) {
```
Comment (pdex) [inline src/main/java/com/xjeffrose/chicago/db/EncryptedStorageProvider.java]: package local please

File: src/main/java/com/xjeffrose/chicago/db/EncryptedStorageProvider.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+      return os.toByteArray();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    return null;
+  }
+
+  public byte[] decrypt(byte[] encryptedData) {
```
Comment (pdex) [inline src/main/java/com/xjeffrose/chicago/db/EncryptedStorageProvider.java]: package local please

File: src/main/java/com/xjeffrose/chicago/db/InMemDBImpl.java [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
 import java.util.Map;
 import java.util.concurrent.atomic.AtomicLong;
 import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @Slf4j
-public class InMemDBImpl implements DBInterface, AutoCloseable {
+public class InMemDBImpl implements StorageProvider, AutoCloseable {
+  private static final Logger log = LoggerFactory.getLogger(InMemDBImpl.class);
```
Comment (pdex) [inline src/main/java/com/xjeffrose/chicago/db/InMemDBImpl.java]: use the lombok please

File: src/main/java/com/xjeffrose/chicago/db/StorageProvider.java [hunk 4]
```diff
@@ -17,4 +17,6 @@
   byte[] stream(byte[] colFam, byte[] key);
 
   byte[] tsWrite(byte[] colFam, byte[] key, byte[] val);
+
+  void stop();
```
Comment (pdex) [inline src/main/java/com/xjeffrose/chicago/db/StorageProvider.java]: Should we extend AutoCloseable instead of defining stop?

Comment (pdex) [inline src/main/java/com/xjeffrose/chicago/db/StorageProvider.java]: Should we have an open as well?

File: src/main/java/com/xjeffrose/chicago/server/ChicagoServer.java [hunk 5]
```diff
@@ -16,17 +20,32 @@
...
     zkClient = new ZkClient(config.getZkHosts(),true);
     rocksDbImpl = new RocksDBImpl(config);
     nodeWatcher = new NodeWatcher(NODE_LIST_PATH, NODE_LOCK_PATH, config.getQuorum());
-    dbRouter = new DBRouter(config, rocksDbImpl, dbLog);
+    dbRouter = new DBRouter(getStorageProvider(config));
 //    config.setZkClient(zkClient);
   }
 
+  private StorageProvider getStorageProvider(ChiConfig config) {
+    if (config.isEncryptAtRest()) {
```
Comment (pdex) [inline src/main/java/com/xjeffrose/chicago/server/ChicagoServer.java:34]: Should we turn this into an enum at some point?

File: src/main/java/com/xjeffrose/chicago/server/DBRouter.java [hunk 6]
```diff
@@ -1,206 +1,64 @@
...
-    XioServerDef dbServer = new XioServerDefBuilder()
-        .name("Chicago DB Server")
-        .listen(new InetSocketAddress(config.getDBBindIP(), config.getDBPort()))
-//        .withSecurityFactory(new XioNoOpSecurityFactory())
-        .withSecurityFactory(new XioSecurityFactory() {
-          @Override
-          public XioSecurityHandlers getSecurityHandlers(XioServerDef xioServerDef, XioServerConfig xioServerConfig) {
-            return new XioSecurityHandlerImpl(config.getCert(), config.getKey());
-          }
+  private final ChicagoPaxosHandler chicagoPaxosHandler;
```
Comment (pdex) [inline src/main/java/com/xjeffrose/chicago/server/DBRouter.java:28]: Should we move all of this into ChicagoServer at some point?

File: src/main/java/com/xjeffrose/chicago/server/DBRouter.java [hunk 7]
```diff
@@ -213,53 +71,15 @@ public void run() {
...
   @Override
   public void close() throws IOException {
     application.close();
-    /*
-    x.stop();
-    serverDefSet.clear();
-    */
     manager.stopAsync().awaitTerminated();
-    rocksDbImpl.destroy();
+    db.stop();
```
Comment (pdex) [inline src/main/java/com/xjeffrose/chicago/server/DBRouter.java]: Should DBManager do this as part of it's shutdown?

File: src/test/java/com/xjeffrose/chicago/InMemDBImplTest.java [hunk 8]
```diff
@@ -1,7 +1,7 @@
 package com.xjeffrose.chicago;
 
 import com.google.common.primitives.Longs;
-import com.xjeffrose.chicago.server.InMemDBImpl;
+import com.xjeffrose.chicago.db.InMemDBImpl;
```
Comment (pdex) [inline src/test/java/com/xjeffrose/chicago/InMemDBImplTest.java:4]: move this test to the db package?

File: src/test/java/com/xjeffrose/chicago/RocksDBImplTest.java [hunk 9]
```diff
@@ -5,7 +5,7 @@
 import com.typesafe.config.Config;
 import com.typesafe.config.ConfigFactory;
 import com.xjeffrose.chicago.server.ChiConfig;
-import com.xjeffrose.chicago.server.RocksDBImpl;
+import com.xjeffrose.chicago.db.RocksDBImpl;
```
Comment (pdex) [inline src/test/java/com/xjeffrose/chicago/RocksDBImplTest.java:8]: move this test to the db package?

File: src/test/resources/reference.conf [hunk 10]
```diff
@@ -39,8 +39,8 @@ chicago {
           bindPort = 12001
         }
       }
-      rpc = ${xio.serverTemplate} {
-        name = "Chicago RPC Server"
+      paxos = ${xio.serverTemplate} {
```
Comment (pdex) [inline src/test/resources/reference.conf:42]: Did we forget to deprecate this file? We should be using `src/main/resources/reference.conf`

File: src/main/java/com/xjeffrose/chicago/server/ChicagoServer.java (hunk 5 above)
Comment (xjdr) [inline src/main/java/com/xjeffrose/chicago/server/ChicagoServer.java:34]: I agree

File: src/main/java/com/xjeffrose/chicago/server/DBRouter.java (hunk 6 above)
Comment (xjdr) [inline src/main/java/com/xjeffrose/chicago/server/DBRouter.java:28]: I think the class is the correct place for this, but the name of the class is probably misleading.

Comment (pdex) [inline src/main/java/com/xjeffrose/chicago/server/DBRouter.java:28]: Let's make this a TODO item?

File: src/main/java/com/xjeffrose/chicago/server/ChicagoServer.java (hunk 5 above)
Comment (pdex) [inline src/main/java/com/xjeffrose/chicago/server/ChicagoServer.java:34]: TODO item'd

File: src/test/resources/reference.conf (hunk 10 above)
Comment (xjdr) [inline src/test/resources/reference.conf:42]: test doesn't work without this file

File: src/main/java/com/xjeffrose/chicago/server/DBRouter.java (hunk 6 above)
Comment (xjdr) [inline src/main/java/com/xjeffrose/chicago/server/DBRouter.java:28]: Done

**Pull request merged**